### PR TITLE
Consider switching from Jekyll to MkDocs

### PR DESCRIPTION
**Pull request by bradyt**
This pull request is in regards to discussion at https://github.com/xuhcc/beancount-docs/issues/3, with proposed changes to make the switch.

Switching from Jekyll to MkDocs would provide more navigation features then we currently see, including tab bar navigation and table of contents for each page (beyond that of the toc generated from Google Docs source, it has a toc that is on left pane or above text, depending on screen dimensions). It also provides a search feature, and an edit page button.

The changes below first add a MkDocs config file, which enables the tab bar navigation, and footnotes. The header anchors are removed, so that MkDocs toc is generated. The image paths are fixed, so that images render. Finally a repo_url is set, so that the edit page button works.

A live example can be previewed at https://bradyt.github.io/beancount-docs/.

Local preview can be tested with `mkdocs serve`.

Site can be deployed with `mkdocs gh-deploy`.

### CODE REVIEW

**Comment by bradyt** [comment]
Github apparently cannot tell the order of commits above, despite commits containing information of parent. I would recommend fetching the changes to clarify.

File: filter.py [hunk 1]
```diff
@@ -97,8 +97,6 @@ def action(elem, doc):
         # There must be only one level 1 header
         if elem.identifier != 'title':
             elem.level += 1
-        # Add explicit anchor
-        elem.content.insert(0, RawInline(f'<a id="{elem.identifier}"></a>'))
```
Comment (xuhcc) [inline filter.py]: Is there a way to make TOC working with tags?

Without anchors, navigation breaks on github itself. For example, the "Credits & Debits" link here is not working: https://github.com/bradyt/beancount-docs/blob/mkdocs/docs/02_the_double_entry_counting_method.md

Comment (xuhcc) [reply filter.py]: It seems to be working if we add anchor after the heading:
```
elem.content.append(RawInline(f'<a id="{elem.identifier}"></a>'))
```

File: filter.py [hunk 2]
```diff
@@ -113,6 +111,9 @@ def action(elem, doc):
         # Remove unnecessary leading tabs from code blocks
         elem.text = re.sub(r'^\t', '', elem.text, flags=re.MULTILINE)
 
+    elif isinstance(elem, Image):
+        elem.url = './' + elem.url
```
Comment (xuhcc) [inline filter.py]: Similar problem here, github doesn't show images: https://github.com/bradyt/beancount-docs/blob/mkdocs/docs/02_the_double_entry_counting_method.md#basics-of-double-entry-bookkeeping

I found that setting `use_directory_urls: false` in `mkdocs.yml` can help:

https://github.com/mkdocs/mkdocs/blob/master/docs/user-guide/configuration.md#use_directory_urls

File: mkdocs.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        - Beancount Scripting Plugins: 23_beancount_scripting_plugins.md
+        - Beancount Design Doc: 24_beancount_design_doc.md
+        - Ledgerhub Design Doc: 25_ledgerhub_design_doc.md
+        - External Contributions: 26_external_contributions.md
+      - Enhancement Proposals & Discussions:
+        - A Proposal for an Improvement on Inventory Booking: 27_a_proposal_for_an_improvement_on_inventory_booking.md
+        - Settlement Dates in Beancount: 28_settlement_dates_in_beancount.md
+        - Balance Assertions in Beancount: 29_balance_assertions_in_beancount.md
+        - Fund Accounting with Beancount: 30_fund_accounting_with_beancount.md
+        - Rounding Precision in Beancount: 31_rounding_precision_in_beancount.md
```
Comment (xuhcc) [inline mkdocs.yml]: Can we put this list outside of mkdocs.yml? I would like to generate it automatically.

**Comment by bradyt** [comment]
I don't mind if you don't use any of it. I'm not sure when I can work on this again. It was mostly just to show that the wiki can work for some configuration. If you find ways to minimize hand generated code, or make things work on both mkdocs wiki _and_ github interface, that sounds excellent and you should replace commits if you can.

**Comment by xuhcc** [comment]
@bradyt I cherry-picked two commits from your branch and added aforementioned fixes on top of them. Thanks for your contribution!

**Pull request closed without merging**